Sources/SwiftProcessing/Core/UIKitElements/Capture.swift

```diff
@@ -0,0 +1,385 @@
+        
+        let newImage = UIGraphicsGetImageFromCurrentImageContext()
+        
+        UIGraphicsEndImageContext()
+        
+        return newImage
+    }
+    
+    
+    func prepare(
+        cameraPosition: AVCaptureDevice.Position,
+        desiredFrameRate: Int? = nil,
+        completion: ((Bool) -> Void)?
+    ) {
+        
+        self.preparingCompletionHandler = completion
+        self.cameraPosition = cameraPosition
+        checkCameraAccess { allowed in
+            if allowed { self.setup(desiredFrameRate) }
+            completion?(allowed)
+            self.preparingCompletionHandler = nil
+        }
+    }
+    
+    private func setup(_ desiredFrameRate: Int? = nil) {
+        configureCaptureSession(desiredFrameRate)
+    }
+    
+    func start() {
+        if cameraIsReadyToUse { session.startRunning() }
+    }
+    
+    func stop() {
+        session.stopRunning()
+    }
+    
+}
+
+extension Camera {
+    
+    private func askUserForCameraPermission(_ completion:  ((Bool) -> Void)?) {
+        AVCaptureDevice.requestAccess(for: AVMediaType.video) { (allowedAccess) -> Void in
+            DispatchQueue.main.async { completion?(allowedAccess) }
+        }
+    }
+    
+    private func checkCameraAccess(_ completion: ((Bool) -> Void)?) {
+        askUserForCameraPermission { [weak self] allowed in
+            guard let self = self, let completion = completion else { return }
+            self.cameraIsReadyToUse = allowed
+            if allowed {
+                completion(true)
+            } else {
+                self.showDisabledCameraAlert(completion)
+                print("No Access to Camera")
+            }
+        }
+    }
+    
+    private func configureCaptureSession(_ desiredFrameRate: Int? = nil) {
+        guard let previewView = previewView else { return }
+        // Define the capture device we want to use
+        
+        guard let camera = AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: cameraPosition) else {
+            let error = NSError(domain: "", code: 0, userInfo: [NSLocalizedDescriptionKey : "No front camera available"])
+            let errorAlert = Alert(error)
+            errorAlert.show()
+            return
+        }
+        
+        // Connect the camera to the capture session input
+        do {
+            
+            try camera.lockForConfiguration()
+            defer { camera.unlockForConfiguration() }
+            
+            if camera.isFocusModeSupported(.continuousAutoFocus) {
+                camera.focusMode = .continuousAutoFocus
+            }
+            
+            if camera.isExposureModeSupported(.continuousAutoExposure) {
+                camera.exposureMode = .continuousAutoExposure
+            }
+            if desiredFrameRate != nil {
+                
+                guard let range = camera.formats.first!.videoSupportedFrameRateRanges.first,
```

Comment:
210 -> 226 move that to private function named something like configureFrameRate